### PR TITLE
Wrap sidecar in arcs

### PR DESCRIPTION
**Pull request by kdonthi**
ref #11460

### CODE REVIEW

File: crates/transaction-pool/src/lib.rs [hunk 1]
```diff
@@ -196,6 +196,8 @@ pub mod identifier;
 mod ordering;
 mod traits;
 
+type SharedBlobTransactionSidecar = Arc<BlobTransactionSidecar>;
```
Comment (mattsse) [inline crates/transaction-pool/src/lib.rs:199]: I don't really want an alias for this, so I'd like to remove this

File: crates/transaction-pool/src/blobstore/disk.rs [hunk 2]
```diff
@@ -279,15 +280,17 @@ impl DiskFileBlobStoreInner {
...
-    fn get_one(&self, tx: B256) -> Result<Option<BlobTransactionSidecar>, BlobStoreError> {
+    fn get_one(&self, tx: B256) -> Result<Option<Arc<BlobTransactionSidecar>>, BlobStoreError> {
         if let Some(blob) = self.blob_cache.lock().get(&tx) {
             return Ok(Some(blob.clone()))
         }
         let blob = self.read_one(tx)?;
+
         if let Some(blob) = &blob {
-            self.blob_cache.lock().insert(tx, blob.clone());
+            self.blob_cache.lock().insert(tx, Arc::new(blob.clone()));
```
Comment (mattsse) [inline crates/transaction-pool/src/blobstore/disk.rs:290]: this should be Arc'ed here, otherwise we clone this twice

**Review by mattsse — CHANGES_REQUESTED** [review]
this is looking great already.

File: crates/transaction-pool/src/blobstore/disk.rs (hunk 2 above)
Comment (kdonthi) [reply crates/transaction-pool/src/blobstore/disk.rs:289]: I am not sure what this means 😕

**Review by mattsse — APPROVED** [review]
ty, lgtm

need to do one final check before merging

**Comment by mattsse** [comment]
does not compile yet though, looks like one function needs to be migrated

**Comment by kdonthi** [comment]
🤞

File: testing/testing-utils/src/generators.rs [hunk 3]
```diff
@@ -323,7 +323,7 @@ where
                 let old = if entry.value.is_zero() {
                     let old = storage.remove(&entry.key);
                     if matches!(old, Some(U256::ZERO)) {
-                        return None
+                        return None;
```
Comment (Rjected) [inline testing/testing-utils/src/generators.rs:326]: can we just remove the semicolon here

**Review by Rjected — COMMENTED** [review]
would like one smol change, otherwise lgtm

File: testing/testing-utils/src/generators.rs (hunk 3 above)
Comment (kdonthi) [reply testing/testing-utils/src/generators.rs:326]: no promises the linter will like it 😛

Comment (kdonthi) [reply testing/testing-utils/src/generators.rs]: Done

**Review by Rjected — APPROVED** [review]
lgtm

**Review by mattsse — APPROVED** [review]
lgtm,

sorry about the delay.

I applied some updates: `Arc::unwrap_or_clone` which makes the (*).clone a bit nicer

**Pull request merged**